### PR TITLE
TE-535: Changed globby to async approach

### PR DESCRIPTION
**Pull request by supproduction**
https://spryker.atlassian.net/browse/TE-535

### CODE REVIEW

File: lib/build.js [hunk 1]
```diff
@@ -4,61 +4,65 @@ const colors = require('colors/safe');
...
-        log.error(colors.red('webpack configuration:'), error.toString());
-        log.error('build aborted');
-
-        if (callback) {
-            callback(error);
         }
-    }
+    }, error => {
+        log.error(error, 'imposible to resolve configuration promise');
+    })
```
Comment (abelspry) [inline lib/build.js]: I think this is overkill. Why the builder should take a promise in input? What if our customer decides to change the building system and use a normal config?
I cannot allow this, it's too contextual and does not take in account possible consequences on customer usage or existing configurations.
Revert the change and adopt a different solution, maybe using an await before the call to build and passing an actual configuration to webpack.

File: lib/find.js [hunk 2]
```diff
@@ -11,26 +11,36 @@ function parseSettings(settings) {
...
 }
 
 function parseOrigin(initial) {
     return initial || {};
 }
 
-function glob(configuration, dirs, patterns) {
-    return dirs.reduce((results, dir) => {
+async function globbyAsync(patterns, rootConfiguration) {
+    let rootResults = await globby(patterns, rootConfiguration);
```
Comment (abelspry) [inline lib/find.js]: 1. this could be a `const` but
2. return directly the value without unnecessary assignment.

File: lib/find.js [hunk 3]
```diff
@@ -11,26 +11,36 @@ function parseSettings(settings) {
...
-        let rootResults = globby
-            .sync(patterns, rootConfiguration)
-            .map(result => path.join(dir, result));
+        return await globbyAsync(patterns, rootConfiguration).then(globbyResult => {
+            const globbyFullPath = globbyResult.map(result => path.join(dir, result));
+
+            return results.concat(globbyFullPath);
+        }, error => {
+            log.error(error, 'impossible to resolve globby promise');
+        })
```
Comment (abelspry) [inline lib/find.js]: can we better refactor here? We have an await in an await. Please create 2 functions instead and make this code cleaner.

Comment (abelspry) [reply lib/find.js]: Please provide better error message explaining what it means `globby promise`, that from a customer prospective has no meaning.
I guess it's `An error occurred while globbing the system for entry points.`

File: lib/build.js (hunk 1 above)
Comment (supproduction) [reply lib/build.js]: Fixed

File: lib/find.js (hunk 2 above)
Comment (supproduction) [reply lib/find.js]: Fixed

File: lib/find.js (hunk 3 above)
Comment (supproduction) [reply lib/find.js]: Fixed. Made function cleaner

**Pull request merged**